app/controllers/home_controller.rb

```diff
@@ -4,8 +4,25 @@ class HomeController < ApplicationController
   def index
     @rentals = Rental.all
     @item_types = ItemType.all
-    @upcoming_rentals = @rentals.where('user_id = ? AND end_time >= ?', @current_user.id, Time.zone.today)
-    @q = @rentals.where('user_id = ? AND end_time < ?', @current_user.id, Time.zone.today).search(params[:q])
+    categorize_rentals
+    non_admin_filter unless @current_user.groups.find_by(name: 'admin').present?
+  end
+
+  private
+
+  def categorize_rentals
+    @upcoming_rentals = @rentals.where('end_time > ? AND rental_status = ?', DateTime.current, 'reserved')
+    @ongoing_rentals = @rentals.where('rental_status = ?', 'checked_out')
+    @no_show_rentals = @rentals.where('end_time <= ? AND rental_status = ?', DateTime.current, 'reserved')
```

Comment:
Four(ish) critiques with these queries. We also did not do a sufficient job explaining who is using the index page, and for what reason.

1. Mea Culpa for not framing the use case. The index page will be used by Admins to view upcoming rentals for that day, or maybe today and tomorrow. Like a dashboard of customers who could be showing up, throughout the day. So, if we have a couple hundred reservations (on future date) those rentals will show up in `@upcoming_rentals`. Please amend your first query to look for rentals starting today, in the reserved status. HINT: You would be able to accomplish using the hash syntax `.where(key: value. key_two: value_two)`
2. I would strive for having the model pass the state values to you, as opposed to hard coding the strings. Double check the [gem documentation](https://github.com/aasm/aasm) to see if you can get the state names from method calls. Follow?
3. Double check to see if we are using similar lookups other places. If we are, I think this is a great target for a scope!
4. Using pure string conditions is not frowned upon, sometimes (like when using straight inequalities) you have to use it. However, for a variety of reasons [including security](http://guides.rubyonrails.org/active_record_querying.html#pure-string-conditions), using hash conditionals is much preferred by me. Further, you can often get away with using date arrays (for example), if you are trying to select a range. Lastly. remember if you ever get really stuck. there is not thing that prevents chaining `where`s as in: 'Model.where(key: value).where('date > ?', value)'